crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -380,6 +380,23 @@ where
     marker: PhantomData<fn() -> Marker>,
 }
 
+// De-initializes the cloned system.
+impl<Marker, F> Clone for FunctionSystem<Marker, F>
+where
+    F: SystemParamFunction<Marker> + Clone,
```

Comment:
This behavior, while probably correct, is very subtle. It needs to be documented somewhere visible -- perhaps on the `System` trait? Or, maybe it should be documented on `distributive_run_if`, since I think that's the only place we're cloning systems right now.

Reply:
Right now `Clone` is implemented specifically for `FunctionSystem`, and not necessarily for other `System` implementers. I think it makes more sense to call it out on `FunctionSystem`. I also added a note on `distributive_run_if`.

Reply:
> Right now Clone is implemented specifically for FunctionSystem, and not necessarily for other System implementers.

Right, but users will rarely if ever interact with the `FunctionSystem` type itself. I think it's more likely to be seen if the `System` trait mentions that systems defined with a `fn` have this property.

Reply:
Why is this resolved?

Reply:
This was addressed below in https://github.com/bevyengine/bevy/pull/8105/commits/3a98270f45b750d49fa84adbbbbdeb7e1bd33471.

Reply:
That documented it on `FunctionSystem`, but users have little reason to ever interact with that type or read its documentation. Function systems get automagically created through type magic, so many users are probably only vaguely aware of that type's existence because of compiler errors.